src/control/Control.Layers.js

```diff
@@ -187,24 +188,22 @@ export var Layers = Control.extend({
 		if (collapsed) {
 			this._map.on('click', this.collapse, this);
 
-			if (!Browser.android) {
-				DomEvent.on(container, {
-					mouseenter: this.expand,
-					mouseleave: this.collapse
-				}, this);
-			}
+			DomEvent.on(container, {
+				mouseenter: function () {
+					this._deferredExpand = setTimeout(Util.bind(this.expand, this));
+				},
+				mouseleave: this.collapse
+			}, this);
 		}
 
 		var link = this._layersLink = DomUtil.create('a', className + '-toggle', container);
 		link.href = '#';
 		link.title = 'Layers';
 
-		if (Browser.touch) {
-			DomEvent.on(link, 'click', DomEvent.stop);
-			DomEvent.on(link, 'click', this.expand, this);
-		} else {
-			DomEvent.on(link, 'focus', this.expand, this);
-		}
+		DomEvent.on(link, 'click', function (e) { // necessary for touch
+			DomEvent.preventDefault(e); // prevent link function
+			this.expand();
+		}, this);

```

Comment:
```suggestion
		DomEvent.on(link, 'focus', this.expand, this); // necessary for touch
		DomEvent.on(link, 'click', DomEvent.preventDefault); // prevent link function
```

I'm not sure if we really need react to `focus` event though, but that was in original code.
Thoughts?